### PR TITLE
RATEST-181: Implement 3.x User settings workflow

### PR DESCRIPTION
**Pull request by jayasanka-sack**
## Purpose
The purpose of this PR is to fix: https://issues.openmrs.org/projects/RATEST/issues/RATEST-181

## Goals
To give users confidence that the Settings workflow of the RefApp 3.x is working as expected.

## Approach
- Implement the steps for the settings workflow
- Replace the properties with environment variables
- Implement commands to register and delete a patient
- Upgrade the cypress version

## Workflow Case

- (as logged-in user)
- (on a patient’s chart)
- Navigate back to home by clicking Home breadcrumb
- The user can change the locale to Spanish and the text will change
- Location: 
   - Display current location
   - Change location: The user can change their login location, and find themselves back on the patient’s chart
- Log out: 
   - The user can log out and find themselves back on the login page

### CODE REVIEW

**Comment by HerbertYiga** [comment]
thanks @jayasanka-sack  are you able to figure out what is breaking the  3.x user settings build?

**Comment by jayasanka-sack** [comment]
@HerbertYiga It was a typo on a step implementation. Fixed it in https://github.com/openmrs/openmrs-contrib-qaframework/pull/64/commits/066488d550a688c57f712323027aafc641c2f595

File: .github/workflows/refapp-3x-login.yml
Comment (brandones) [inline .github/workflows/refapp-3x-login.yml:21]:
```suggestion
      - name: Run user settings workflow tests
```
?

File: cypress/support/commands.js
Comment (brandones) [inline cypress/support/commands.js:85]: That's odd. Does the `null` identifier cause it to auto-generate? If so, please leave a comment to that effect.

**Review by brandones — APPROVED** [review]
One question, one suggestion, but generally—this is really nice work. Well done, and thank you.

File: .github/workflows/refapp-3x-login.yml
Comment (jayasanka-sack) [reply .github/workflows/refapp-3x-login.yml:21]: No. This is the workflow file of the login. There was a typo, and I fixed it.

File: cypress/support/commands.js
Comment (jayasanka-sack) [reply cypress/support/commands.js:85]: No, actually it's being replaced with the retrieved identifier from the `iden` endpoint. However, it's clear that there's a readability issue in the code. I'll add a comment.

**Comment by brandones** [comment]
Looks like we've got a build error. Otherwise this is good to merge.

**Review by kdaud — APPROVED** [review]
@brandones the commit LGTM . 
Feel free to make further recommendation !

**Comment by sherrif10** [comment]
@kdaud  , its always good for these kind of tests to first let @brandones  to give final recommendation

**Comment by kdaud** [comment]
> its always good for these kind of tests to first let @brandones to give final recommendation

FYI I and @brandones are mentors for the GSoC project of  Automated Tests for [3.x RefApp](https://talk.openmrs.org/t/e2e-automated-tests-for-the-openmrs-3-0-refapp/33702) which @jayasanka-sack has been developing through out this summer.  He is in his final week and would like to see him succeed. 

I took time to test the workflow locally and its passing and @brandones had approved the PR [here](https://talk.openmrs.org/t/e2e-automated-tests-for-the-openmrs-3-0-refapp/33702) for which the student addressed thetest failures in the PRs that followed his approval. Well @brandones have done the merge though he seems to be occupied and waiting for him would be ok but his approval is good enough to aid the student progress and BTW few days to the end of GSoC-2021 

cc: @sherrif10 @brandones

**Comment by brandones** [comment]
Yep, I said good to merge once build errors are fixed, and build errors were fixed, so we're all good.

**Pull request merged**
